lib/thor/parser/options.rb

```diff
@@ -45,6 +45,7 @@ def initialize(hash_options = {}, defaults = {}, stop_on_unknown = false, disabl
       @switches = {}
       @extra = []
       @stopped_parsing_after_extra_index = nil
+      @is_switch_value = false
```

Comment:
This is actually incorrect, no? This is a value with a dash that is specifically *not* a switch.

Reply:
I'm not sure I understand. This instance variable is `true` if and only if the current element in the pile being processed is considered to be a value _even if_ it has a leading `-`. The initial state is that nothing has been parsed yet, in which case we want to parse it as an option if it has a leading `-`, not as a value.

Reply:
Yep, exactly. So when this variable is true, the current argument is a value, *not* a switch. The name "is switch value" seems to say "the current value is actually a switch".

Maybe something like `is_literal_value` or something similar?

Reply:
I landed on `is_treated_as_value`. PTAL @dorner